src/path.jl

```diff
@@ -121,23 +121,31 @@ Base.parent(fp::AbstractPath) = parents(fp)[end]
 """
     parents{T<:AbstractPath}(fp::T) -> Array{T}
 
+Return all parents of the path. If no parent exists then either "/" or "."
+will be returned depending on whether the path is absolute.
+
 # Example
 ```
 julia> parents(p"~/.julia/v0.6/REQUIRE")
 3-element Array{FilePathsBase.PosixPath,1}:
  p"~"
  p"~/.julia"
  p"~/.julia/v0.6"
- ```
 
-# Throws
-* `ErrorException`: if `path` doesn't have a parent
+julia> parents(p"/etc")
+1-element Array{PosixPath,1}:
+ p"/"
+
+julia> parents(p"etc")
+1-element Array{PosixPath,1}:
+ p"."
+ ```
 """
 function parents(fp::T) where {T <: AbstractPath}
     if hasparent(fp)
         return [Path(fp, fp.segments[1:i]) for i in 1:length(fp.segments)-1]
     else
-        error("$fp has no parents")
+        return [isempty(fp.root) ? Path(fp, tuple(".")) : Path(fp, ())]
```

Comment:
If possible, can we make it so that the result of 

```
parents(x) === x
```
for this case?

Reply:
Just so we aren't allocating?

Reply:
Because in similar cases (e.g. array parents)
I do
`isroot(x) = parent(x)===x

Reply:
Hmmm, I think that's a bit weird, but sure. I'll note that your `isroot` method is wrong because it would return true for even non-root relative paths (e.g., `isroot(p".")` should probably return false).